service/population.go

```diff
@@ -20,29 +21,36 @@ func NewPopulationService(populationRepository repository.PopulationRepository)
 	}
 }
 
-func (p *populationService) GetPopulationStatistics() ([]model.PopulationResponseItem, error) {
-	out := []model.PopulationResponseItem{}
+func (p *populationService) generateLogger(functionName string) *logrus.Entry {
+	return logrus.WithFields(logrus.Fields{
+		"Module":  "Service",
+		"Funtion": functionName,
+	})
+}
+
+func (p *populationService) GetPopulationStatistics() (populationStatistic []model.PopulationResponseItem, err error) {
 	districts, err := p.repository.QueryAllDistrict()
 	if err != nil {
-		logrus.Error(err)
+		p.generateLogger("QueryAllDistrict").Error(err)
+		return []model.PopulationResponseItem{}, err
 	}
 	for _, s := range districts {
 		districtId := s.DistrictID
 		districtName := s.Name
 		total, err := p.repository.QueryTotalPopulation(districtId)
 		if err != nil {
-			logrus.Error(err)
-			continue
+			p.generateLogger("QueryTotalPopulation").Error(err)
```

Comment:
Generate logger only once for the same function name.

Reply:
fixed